From a303548f8a674f8ef43607f0e134574968fa57d2 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Sun, 31 Mar 2024 07:31:11 -0400
Subject: [PATCH] arm64: Kconfig: Don't set DMA_BOUNCE_UNALIGNED_KMALLOC

If enabled, DMA_BOUNCE_UNALIGNED_KMALLOC causes the swiotlb buffers
(64MB, by default) to be allocated, even on systems where the DMA
controller can reach all of RAM.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm64/Kconfig | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index f9777ce2ccb2..c9691b0c0f29 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -122,7 +122,8 @@ config ARM64
 	select CRC32
 	select DCACHE_WORD_ACCESS
 	select DYNAMIC_FTRACE if FUNCTION_TRACER
-	select DMA_BOUNCE_UNALIGNED_KMALLOC
+	# Disable this to save 64MB when DMA controllers can reach all of RAM
+	# select DMA_BOUNCE_UNALIGNED_KMALLOC
 	select DMA_DIRECT_REMAP
 	select EDAC_SUPPORT
 	select FRAME_POINTER
-- 
2.39.2

